File: Matury CKE/2022 Maj/Programowanie/Python/zad4_3.py
#numbers=[int(l.strip()) for l in open('../../Dane_2205/przyklad.txt').readlines()]
numbers=[int(l.strip()) for l in open('../../Dane_2205/liczby.txt').readlines()]

# a) Ile jest dobrych trójek (trójki wraz z zapisaniem do pliku trojki.txt)?

goodThreeCount=0
trojki=open('./trojki.txt', 'w')

for i in numbers:
    for j in numbers:
        if j%i>0 or i==j:
            continue
        for k in numbers:
            if i==k or j==k:
                continue
            if k%j==0:
                trojki.write(f'{str(i)} {str(j)} {str(k)}\n')
                goodThreeCount+=1
trojki.close()

# a) Ile jest dobrych piątek?

goodFiveCount=0

for i in numbers:
    for j in numbers:
        if i==j:
            continue
        if j%i>0:
            continue
        for k in numbers:
            if i==k or j==k:
                continue
            if k%j>0:
                continue
            for l in numbers:
                if i==l or j==l or k==l:
                    continue
                if l%k>0:
                    continue
                for m in numbers:
                    if i==m or j==m or k==m or l==m:
                        continue
                    if m%l==0:
                        goodFiveCount+=1

print('4.3.', 'a)', goodThreeCount, 'b)', goodFiveCount)

File: Matury CKE/2022 Maj/Programowanie/C++/zad4_3.cpp
#include <iostream>
#include <fstream>
#include <string>
#include <vector>

int main()
{
    std::cout<<"4.3. ";

    // std::ifstream dane("C:/Matury-Informatyka/Matury CKE/2022 Maj/Dane_2205/przyklad.txt");
    std::ifstream dane("C:/Matury-Informatyka/Matury CKE/2022 Maj/Dane_2205/liczby.txt");

    std::ofstream trojki("trojki.txt");

    std::vector<int> numbersList;

    int number;
    while(dane >> number){
        numbersList.push_back(number);
    }

    int trojkiCount = 0;

    for(auto x: numbersList){
        for(auto y: numbersList){
            if(x==y)continue;
            if(y%x!=0)continue;
            for(auto z: numbersList){
                if(y==z)continue;
                if(z%y!=0)continue;
                trojkiCount+=1;
                trojki << x << " " << y << " " << z << std::endl;
            }
        }
    }

    std::cout << "a) " << trojkiCount << " ";

    trojki.close();

    int piatkiCount = 0;

    for(auto u: numbersList){
        for(auto w: numbersList){
            if(u==w)continue;
            if(w%u!=0)continue;
            for(auto x: numbersList){
                if(w==x || u==x)continue;
                if(x%w!=0)continue;
                for(auto y: numbersList){
                    if(x==y || w==y || u==y)continue;
                    if(y%x!=0)continue;
                    for(auto z: numbersList){
                        if(y==z || x==z || w==z || u==z)continue;
                        if(z%y!=0)continue;
                        piatkiCount += 1;
                    }
                }
            }
        }
    }

    std::cout<<"b) " << piatkiCount;

    return 0;
}
